From c40c8220fa67b963da061aaa3fb96e2191932f44 Mon Sep 17 00:00:00 2001
From: "Kevin F. Haggerty" <haggertk@lineageos.org>
Date: Sun, 21 Oct 2018 08:21:25 -0600
Subject: [PATCH 03/19] msm8974-common: sepolicy: Label sysfs_graphics nodes

* Migrate sysfs_mdnie to sysfs_graphics while we are at it

Change-Id: I8597d7be6217816924a8fee854341e4f2fb18562
---
 sepolicy/common/file.te          |  1 -
 sepolicy/common/file_contexts    | 12 +++++-------
 sepolicy/common/system_app.te    |  2 +-
 sepolicy/common/system_server.te |  2 +-
 4 files changed, 7 insertions(+), 10 deletions(-)

diff --git a/sepolicy/common/file.te b/sepolicy/common/file.te
index 2e2036d..8df7b10 100644
--- a/sepolicy/common/file.te
+++ b/sepolicy/common/file.te
@@ -2,7 +2,6 @@ type proc_bt_sleep, fs_type;
 
 type sysfs_camera, fs_type, sysfs_type;
 type sysfs_hal_pwr, fs_type, sysfs_type;
-type sysfs_mdnie, fs_type, sysfs_type;
 type sysfs_sec, fs_type, sysfs_type;
 type sysfs_wifi_writeable, fs_type, sysfs_type;
 
diff --git a/sepolicy/common/file_contexts b/sepolicy/common/file_contexts
index b844035..0799d46 100644
--- a/sepolicy/common/file_contexts
+++ b/sepolicy/common/file_contexts
@@ -30,16 +30,14 @@
 /data/cam_socket3                                       u:object_r:camera_socket:s0
 
 # sysfs
+/sys/class/mhl(/.*)?                                             u:object_r:sysfs_graphics:s0
 /sys/devices/battery.[0-9]+/power_supply/battery(/.*)?           u:object_r:sysfs_batteryinfo:s0
+/sys/devices/mdp\.[0-9]+(/.*)?                                   u:object_r:sysfs_graphics:s0
 /sys/devices/platform/bcm[0-9]+_bluetooth/rfkill/rfkill0/state   u:object_r:sysfs_bluetooth_writable:s0
 /sys/devices/virtual/camera(/.*)?                                u:object_r:sysfs_camera:s0
+/sys/devices/virtual/graphics(/.*)?                              u:object_r:sysfs_graphics:s0
+/sys/devices/virtual/lcd(/.*)?                                   u:object_r:sysfs_graphics:s0
+/sys/devices/virtual/mdnie(/.*)?                                 u:object_r:sysfs_graphics:s0
 /sys/devices/virtual/sec/sec_key/hall_irq_ctrl                   u:object_r:sysfs_sec:s0
 /sys/module/dhd/parameters/firmware_path                         u:object_r:sysfs_wifi_writeable:s0
 /sys/module/dhd/parameters/nvram_path                            u:object_r:sysfs_wifi_writeable:s0
-
-# mdnie sysfs
-/sys/devices/virtual/lcd/panel/panel/auto_brightness    u:object_r:sysfs_mdnie:s0
-/sys/devices/virtual/mdnie/mdnie/accessibility          u:object_r:sysfs_mdnie:s0
-/sys/devices/virtual/mdnie/mdnie/mode                   u:object_r:sysfs_mdnie:s0
-/sys/devices/virtual/mdnie/mdnie/outdoor                u:object_r:sysfs_mdnie:s0
-/sys/devices/virtual/mdnie/mdnie/scenario               u:object_r:sysfs_mdnie:s0
diff --git a/sepolicy/common/system_app.te b/sepolicy/common/system_app.te
index 5bce4cf..e8ffcf5 100644
--- a/sepolicy/common/system_app.te
+++ b/sepolicy/common/system_app.te
@@ -1 +1 @@
-allow system_app sysfs_mdnie:file rw_file_perms;
+allow system_app sysfs_graphics:file rw_file_perms;
diff --git a/sepolicy/common/system_server.te b/sepolicy/common/system_server.te
index a801e96..7fceb15 100644
--- a/sepolicy/common/system_server.te
+++ b/sepolicy/common/system_server.te
@@ -10,4 +10,4 @@ allow system_server qmuxd_socket:dir w_dir_perms;
 allow system_server qmuxd_socket:sock_file { create setattr write };
 allow system_server qti_debugfs:file r_file_perms;
 allow system_server sensors_device:chr_file r_file_perms;
-allow system_server sysfs_mdnie:file rw_file_perms;
+allow system_server sysfs_graphics:file rw_file_perms;
-- 
2.17.1

